src/app/components/dashboard/dashboard.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, Pipe ,PipeTransform } from '@angular/core';
+import { Observable } from 'rxjs';
+import animateScrollTo from 'animated-scroll-to';
+
+import { ReturnSetInfoDataService } from './dashboard.service';
+
+@Component({
+  selector: 'dashboard',
+  templateUrl: './dashboard.component.html',
+  styleUrls: ['./dashboard.component.scss'],
+  providers: [ ReturnSetInfoDataService ]
+})
+
+@Pipe(
+  {
+    name: "groupByAlphabeticalOrder"
+  }
+)
+export class DashboardComponent{
+  data = []; 
+  index = [];
+  groupedData = [];
+  finalData = [];
+
+  constructor(private dataService: ReturnSetInfoDataService) {
+    dataService.getSetInfoData().subscribe(res => {
+      this.data = res;
+
+      this.data.forEach(gropedItem => {
+        let items = [];
+        if (!gropedItem.name || !gropedItem.name.length) return;
+      
+        let firstLetter = gropedItem.name[0];
+        let index = firstLetter;
+        if (!this.groupedData[firstLetter]) {
+          this.groupedData[firstLetter] = [];
+          this.index.push(firstLetter);
+
+        }
+        
+        this.groupedData[firstLetter].push(gropedItem);       
+  
+      });
+      console.log(this.index);      
+    });
+        
+  }  
+  
+  indexTo(index) {    
```

Comment:
From where did you get this code?

Reply:
@nishintn 
Code for grouping with first letter was taken from a stackoverflow answer (https://stackoverflow.com/questions/34259272/linq-js-grouping-objects-with-there-name-first-letter).
Build other features with this.